### ISSUE TITLE
2 way javascript interop is possible right now?

### ISSUE THREAD

**Issue by danilobreda** [nmoschkin/MAUIWebViewExample#3]
Javascript call C# and C# call Javascript ?

**Comment by nmoschkin** [comment]
#2 

please refer to my answer, here.

**Comment by nmoschkin** [comment]
Okay.  I have created the code for this.  You may test it.

**Comment by nmoschkin** [comment]
Note: Both Windows support and EvaluateJavaScriptAsync support are in the windows-maui-support branch and will be merged, soon.

**Comment by nmoschkin** [comment]
@danilobreda have you had a chance to test it yet?

**Comment by danilobreda** [comment]
@nmoschkin not yet, i will test it soon, thanks for the work :)

**Comment by ibc-shane-kelly** [comment]
@nmoschkin, thanks for this code we are getting further with it than other options out there. I have tested the windows implementation and it is all working well. I am running on iOS now and JavaScript to C# is working but not C# to JavaScript. I will test android tomorrow. I'll let you know what I find.

Quick edit I got the iOS version to work by updating the VirtualView_RequestEvaluateJavaScript function in the iOS HybridWebViewHandler to be

private void VirtualView_RequestEvaluateJavaScript(object sender, EvaluateJavaScriptAsyncRequest e)
        {
            sync.Post((o) =>
            {
                PlatformView.EvaluateJavaScript(e);
            }, null);
        }

Happy to be corrected if there is a better way to do it.

**Comment by ibc-shane-kelly** [comment]
Now tested in Android as well and working well. 

Thanks again!!

**Comment by nmoschkin** [comment]
> @nmoschkin, thanks for this code we are getting further with it than other options out there. I have tested the windows implementation and it is all working well. I am running on iOS now and JavaScript to C# is working but not C# to JavaScript. I will test android tomorrow. I'll let you know what I find.
> 
> Quick edit I got the iOS version to work by updating the VirtualView_RequestEvaluateJavaScript function in the iOS HybridWebViewHandler to be
> 
> private void VirtualView_RequestEvaluateJavaScript(object sender, EvaluateJavaScriptAsyncRequest e) { sync.Post((o) => { PlatformView.EvaluateJavaScript(e); }, null); }
> 
> Happy to be corrected if there is a better way to do it.

yes.  If something fails on iOS, call it from the main thread to see if that fixes it before doing anything else. iOS is very temperamental when it comes to demanding that certain things are always invoked on the main thread.  I did not see this particular issue because at the time, I didn't have a working iOS test platform (this was during that most recent bug that came from upgrading to XCode 14)
